main/drivers/CANDriver.c

```diff
@@ -0,0 +1,293 @@
+
+    // configure MCP251XFD
+    MCP251XFD_Config MCP251863_CAN_conf =
+        DEFAULT_MCP251863_CONTROLLER_CONFIG(&dev->MCP251863_SYSCLK1, &dev->MCP251863_BTStats);
+    MCP251863_CAN_conf.SysInterruptFlags =
+        (MCP251XFD_INT_RX_EVENT | MCP251XFD_INT_RX_OVERFLOW_EVENT | MCP251XFD_INT_CLEARABLE_FLAGS_MASK);
+
+    // Setup FIFO1 for receiving and FIFO2 fro transmitting
+    MCP251XFD_FIFO MCP251863_CAN_FIFO_Conf[CAN_FIFO_COUNT] = {
+        {
+            .Name = MCP251XFD_FIFO1,
+            .Size = MCP251XFD_FIFO_32_MESSAGE_DEEP,
+            .Payload = CAN_RX_PAYLOAD,
+            .Direction = MCP251XFD_RECEIVE_FIFO,
+            .ControlFlags = MCP251XFD_FIFO_NO_TIMESTAMP_ON_RX,
+            .InterruptFlags = MCP251XFD_FIFO_OVERFLOW_INT | MCP251XFD_FIFO_RECEIVE_FIFO_HALF_FULL_INT |
+                              MCP251XFD_FIFO_RECEIVE_FIFO_NOT_EMPTY_INT,
+            .RAMInfos = NULL,
+        }, // SID: 0x000..0x1FF ; No EID
+        {
+            .Name = MCP251XFD_FIFO2,
+            .Size = MCP251XFD_FIFO_4_MESSAGE_DEEP,
+            .Payload = MCP251XFD_PAYLOAD_8BYTE,
+            .Direction = MCP251XFD_TRANSMIT_FIFO,
+            .Attempts = MCP251XFD_THREE_ATTEMPTS,
+            .Priority = MCP251XFD_MESSAGE_TX_PRIORITY16,
+            .ControlFlags = MCP251XFD_FIFO_NO_RTR_RESPONSE,
+            .InterruptFlags = MCP251XFD_FIFO_TX_ATTEMPTS_EXHAUSTED_INT | MCP251XFD_FIFO_TRANSMIT_FIFO_NOT_FULL_INT,
+            .RAMInfos = NULL,
+        },
+    };
+
+    // configure the interupt pin
+    gpio_config_t pinConf = {};
+    pinConf.mode = GPIO_MODE_INPUT;
+    pinConf.intr_type = GPIO_INTR_NEGEDGE;
+    pinConf.pull_down_en = 0;
+    pinConf.pull_up_en = 0;
+    pinConf.pin_bit_mask = (1ULL << interruptPin);
+    gpio_config(&pinConf);
+
+    // Configure standby pin
+    if (standbyPin >= 0) {
+        pinConf.mode = GPIO_MODE_OUTPUT;
+        pinConf.intr_type = GPIO_INTR_DISABLE;
+        pinConf.pin_bit_mask = (1ULL << standbyPin);
+        gpio_config(&pinConf);
+        gpio_set_level(standbyPin, 0); // pull low
+    }
+
+    // Set interrupt handler for the pin.
+    gpio_isr_handler_add(interruptPin, CANDriverISRHandler, (void *)bus);
+    gpio_intr_enable(interruptPin);
+
+    // Configure MCP251863
+    eERRORRESULT ret;
+    ret = MCP251863DeviceSetup(&dev->mcp251863, &MCP251863_CAN_conf, MCP251863_CAN_FIFO_Conf, CAN_FIFO_COUNT);
+
+    // Unmask event
+    driver.deviceEventMask |= (0x1 << bus);
+    xEventGroupSetBits(driver.messageEvents, 1 << bus);
+
+    return ret;
+}
+
+/**
+ * CAN driver update method for periodic tasks, may become obsolete
+ */
+void canDriverUpdate() {
+    //! Unused!
+}
+
+/**
+ * Retrieve a CAN message from the driver.
+ * @param message the CAN message received from the driver
+ * @param timeoutTicks the number of ticks to wait to retrieve a message.
+ * @returns 0 for success or error code
+ */
+int canReceiveMessage(CANMessage_t *message, uint32_t timeoutTicks) {
```

Comment:
This makes it a bit clearer that we're not asking whether the device is able to receive a message. Other functions can be renamed similarly
```suggestion
int receiveCANMessage(CANMessage_t *message, uint32_t timeoutTicks) {
```